cmd/avsc2avdl/main.go

```diff
@@ -0,0 +1,345 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	stdflag "flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"reflect"
+	"sort"
+	"strings"
+
+	"github.com/heetch/avro/internal/typeinfo"
+
+	"github.com/rogpeppe/gogen-avro/v7/schema"
+)
+
+var flag = stdflag.NewFlagSet("", stdflag.ContinueOnError)
+
+func main() {
+	os.Exit(main1())
+}
+
+func main1() int {
+	flag.Usage = func() {
+		fmt.Fprintf(os.Stderr, "usage: avsc2avdl file.avdl\n")
+		flag.PrintDefaults()
+	}
+	if flag.Parse(os.Args[1:]) != nil {
+		return 2
+	}
+	if flag.NArg() != 1 {
+		flag.Usage()
+		return 2
+	}
+	if err := avsc2avdl(flag.Arg(0)); err != nil {
+		fmt.Fprintf(os.Stderr, "avsc2avdl: %v\n", err)
+		return 1
+	}
+	return 0
+}
+
+func avsc2avdl(avscFile string) error {
+	data, err := ioutil.ReadFile(avscFile)
+	if err != nil {
+		return err
+	}
+	at, err := typeinfo.ParseSchema(string(data), nil)
```

Comment:
:tulip: at is a real word so I'd use `avroType` to avoid confusion.
```suggestion
	at, err := typeinfo.ParseSchema(string(data), nil)
```

Reply:
i'm using `at` in lots of places around the `avro` module for an avro type, so i think i'll leave it as is.